beacon-chain/sync/rpc_status.go

```diff
@@ -140,5 +149,11 @@ func (r *RegularSync) validateStatusMessage(msg *pb.Status, stream network.Strea
 	if !bytes.Equal(params.BeaconConfig().GenesisForkVersion, msg.HeadForkVersion) {
 		return errWrongForkVersion
 	}
+	genesis := r.chain.GenesisTime()
+	slotsSinceGenesis := uint64(roughtime.Since(genesis).Seconds()) / params.BeaconConfig().SecondsPerSlot
+	expectedEpoch := helpers.SlotToEpoch(slotsSinceGenesis)
```

Comment:
Expected? Shouldn’t this be max epoch?

Also, the current in progress epoch cannot be finalized until it has completed and been justified, so this doesn’t seem like the correct logic.

Reply:
this is basically just to verify that the given finalized epoch isn't some wild value. I can maybe use the head slot over here ? would be more accurate I guess